<launch>
  <arg name="node_name" default="anbo_node"/>

    <node pkg="dji_osdk_ros" name="$(arg node_name)" type="anbo_node" output="screen">
    <!-- node parameters -->

    <!-- position control -->
    <rosparam name="/waypoints">
    <!-- x, y, z, yaw[deg] -->
    [0.0, 0.0, 5.0, 0.0,
     1.0, 0.0, 0.0, 0.0,
     0.0, 1.0, 0.0, 0.0,
     -1.0, 0.0, 0.0, 0.0,
     0.0, -1.0, 0.0, 0.0,
     
     0.0, 0.0, 0.0, 90.0,
     1.0, 0.0, 0.0, 0.0,
     -1.0, 0.0, 0.0, 0.0,
     0.0, 1.0, 0.0, 0.0,
     0.0, -1.0, 0.0, 0.0,

     0.0, 0.0, 0.0, 90.0,
     1.0, 0.0, 0.0, 0.0,
     -1.0, 0.0, 0.0, 0.0,
     0.0, 1.0, 0.0, 0.0,
     0.0, -1.0, 0.0, 0.0,

     0.0, 0.0, 0.0, 90.0,
     1.0, 0.0, 0.0, 0.0,
     -1.0, 0.0, 0.0, 0.0,
     0.0, 1.0, 0.0, 0.0,
     0.0, -1.0, 0.0, 0.0,

     0.0, 0.0, 0.0, 90.0,

     0.0, 0.0, 8.0, 0.0]</rosparam>
    <param name="/pos_Threshold" type="float" value="0.8"/>
    <param name="/yaw_Threshold" type="float" value="1"/>

    <!-- velocity control -->
    <rosparam name="/velocity_cmd">
    <!-- x, y, z[m/s], yaw[deg/s], time[ms] -->
    [0.0, 0.0, 1.0, 0.0, 5000,
     0.5, 0.0, 0.0, 0.0, 4000,
     0.0, 0.5, 0.0, 0.0, 4000,
     -0.5, 0.0, 0.0, 0.0, 4000,
     0.0, -0.5, 0.0, 0.0, 4000,

     0.0, 0.0, 0.0, 30.0, 3000,
     0.5, 0.0, 0.0, 0.0, 4000,
     -0.5, 0.0, 0.0, 0.0, 4000,
     0.0, 0.5, 0.0, 0.0, 4000,
     0.0, -0.5, 0.0, 0.0, 4000,

     0.0, 0.0, 0.0, 30.0, 3000,
     0.5, 0.0, 0.0, 0.0, 4000,
     -0.5, 0.0, 0.0, 0.0, 4000,
     0.0, 0.5, 0.0, 0.0, 4000,
     0.0, -0.5, 0.0, 0.0, 4000,

     0.0, 0.0, 0.0, 30.0, 3000,
     0.5, 0.0, 0.0, 0.0, 4000,
     -0.5, 0.0, 0.0, 0.0, 4000,
     0.0, 0.5, 0.0, 0.0, 4000,
     0.0, -0.5, 0.0, 0.0, 4000,

     0.0, 0.0, 0.0, 30.0, 3000,

     0.0, 0.0, 1.0, 0.0, 3000]</rosparam>

    </node>
</launch>